src/javascript/expression.rs

```diff
@@ -171,6 +236,35 @@ impl<'module> Generator<'module> {
             "}",
         ))
     }
+
+    fn record_access<'a>(&mut self, record: &'a TypedExpr, label: &'a String) -> Output<'a> {
+        self.not_in_tail_position(|gen| {
+            let record = gen.wrap_expression(record)?;
+            Ok(docvec![record, ".", label])
+        })
+    }
+
+    fn record_update<'a>(
+        &mut self,
+        spread: &'a TypedExpr,
+        updates: &'a Vec<TypedRecordUpdateArg>,
+    ) -> Output<'a> {
+        // println!("{:?}", args);
+        self.not_in_tail_position(|gen| {
+            let spread = gen.wrap_expression(spread)?;
+            let updates: Vec<(Document<'a>, Option<Document<'a>>)> = updates
+                .iter()
+                .map(|TypedRecordUpdateArg {label, value, ..}| Ok((label.to_doc(), Some(gen.wrap_expression(value)?))))
```

Comment:
This and the next few lines is a bit gnarly, but I thin with reusing `wrap_object` and `wrap_args` is at about the right level of abstraction.